iib/workers/tasks/build.py

```diff
@@ -808,24 +808,30 @@ def handle_add_request(
         arches = prebuild_info['arches']
         if deprecation_bundles:
             if is_fbc:
-                err_msg = 'File-Based catalog image type is not supported for deprecation yet.'
-                log.error(err_msg)
-                raise IIBError(err_msg)
-            # opm can only deprecate a bundle image on an existing index image. Build and
-            # push a temporary index image to satisfy this requirement. Any arch will do.
-            arch = sorted(arches)[0]
-            log.info('Building a temporary index image to satisfy the deprecation requirement')
-            _build_image(temp_dir, 'index.Dockerfile', request_id, arch)
-            intermediate_image_name = _get_local_pull_spec(request_id, arch, include_transport=True)
-            deprecate_bundles(
-                deprecation_bundles,
-                temp_dir,
-                prebuild_info['binary_image'],
-                intermediate_image_name,
-                overwrite_from_index_token,
-                # Use podman so opm can find the image locally
-                container_tool='podman',
-            )
+                deprecate_bundles_fbc(
+                    bundles=deprecation_bundles,
+                    base_dir=temp_dir,
+                    binary_image=prebuild_info['binary_image'],
+                    from_index=from_index_resolved,
+                )
+            else:
+                # opm can only deprecate a bundle image on an existing index image. Build and
+                # push a temporary index image to satisfy this requirement. Any arch will do.
+                arch = sorted(arches)[0]
+                log.info('Building a temporary index image to satisfy the deprecation requirement')
+                _build_image(temp_dir, 'index.Dockerfile', request_id, arch)
+                intermediate_image_name = _get_local_pull_spec(
+                    request_id, arch, include_transport=True
+                )
+                deprecate_bundles(
+                    bundles=deprecation_bundles,
+                    base_dir=temp_dir,
+                    binary_image=prebuild_info['binary_image'],
+                    from_index=intermediate_image_name,
```

Comment:
If we are building and pushing an intermediate index, why are we using a local pull spec? 

I was chatting with @chandwanitulsi and I realized there could be a scenario

All the workers use the same virtualenv. What if, when a worker X is processing the request, another request is being processed by a different worker say Y and worker Y runs _cleanup before X gets a chance to use deprecatetruncate on its intermediate image. Because when Y runs cleanup, X's intermediate image would be wiped out, no?

Reply:
1. I think _clenup() in IIB has nothing in common with python virtualenv (if you are referring to that)
2. Clean up runs  `podman rmi --all --force`. I might be wrong here but I thought that this removes all images created by the user that is running the command. And we have multiple users running in parallel but processing requests sequentially. Aren't those images locate in `~/.local/share/containers/storage/` when we are running podman as users and not root?

Reply:
ack, this makes sense too.